app/main.py

```diff
@@ -1,3 +1,17 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Any:
+    cache_dict = {}
+
+    def wrapper(*args) -> Any:
+        if hash(args) in cache_dict:
+            print("Getting from cache")
+            return cache_dict[hash(args)]
+        else:
+            func_res = func(*args)
+            cache_dict[hash(args)] = func_res
+            print("Calculating new result")
+            return func_res
```

Comment:
You can use a single return statement out of the if/else statement instead of two inside it.